Update docs to be sphinx 3+ compatible

## Is your feature request related to a problem? ##
<!-- A clear and concise description of what the problem is. For example, I'm always frustrated when [...] -->

Sphinx is pinned to 1.8.5 which is a pretty old version. I think it has something to do with https://github.com/MDAnalysis/mdanalysis/issues/2667.


## Describe the solution you'd like ##
<!-- A description of what you want to happen. For example, I'd like to be able to do [...] -->

Update the documentation so we don't have to pin sphinx to such an old version. However, we likely do have to pin it to mesh well with the RTD theme -- the user guide recently pinned its versions to `sphinx==3.4.2` and `sphinx_rtd_theme==0.5.1` to fix a search problem. https://github.com/MDAnalysis/UserGuide/pull/141


## Describe alternatives you've considered ##
<!-- A description of any alternative solutions or features you've considered or possible solutions that you've seen elsewhere. -->

Don't and hope for the best.


## Additional context ##
<!-- Add any other context or screenshots about the feature request here. -->

Now our CI doc build **build_docs** is failing, e.g. https://github.com/MDAnalysis/mdanalysis/pull/3421/checks?check_run_id=3690514153

```
running build_sphinx
Running Sphinx v1.8.5
/usr/share/miniconda/envs/test/lib/python3.7/site-packages/setuptools/dist.py:487: UserWarning: Normalizing '2.1.0-dev0' to '2.1.0.dev0'
  warnings.warn(tmpl.format(**locals()))

Warning, treated as error:
Sphinx 1.x is deprecated with sphinx_rtd_theme, update to Sphinx 2.x or greater
Error: Process completed with exit code 1.
```

--- a/.github/workflows/gh-ci.yaml
+++ b/.github/workflows/gh-ci.yaml
@@ -227,7 +227,7 @@ jobs:
 
     - name: build_docs
       run: |
-        cd package && python setup.py build_sphinx -E
+        cd package && python setup.py build_sphinx -E --keep-going
 
     - name: deploy_docs
       if: github.event_name != 'pull_request'

--- a/.github/workflows/gh-ci.yaml
+++ b/.github/workflows/gh-ci.yaml
@@ -217,7 +217,7 @@ jobs:
     - name: install_deps
       run: |
         conda_deps="${{ env.MDA_CONDA_MIN_DEPS }} ${{ env.MDA_CONDA_EXTRA_DEPS}}"
-        pip_deps="${{ env.MDA_PIP_MIN_DEPS}} ${{ env.MDA_PIP_EXTRA_DEPS }} sphinx==1.8.5 sphinx-sitemap sphinx_rtd_theme msmb_theme==1.2.0"
+        pip_deps="${{ env.MDA_PIP_MIN_DEPS}} ${{ env.MDA_PIP_EXTRA_DEPS }} sphinx sphinx-sitemap sphinx_rtd_theme msmb_theme==1.2.0"
         mamba install ${conda_deps}
         pip install ${pip_deps}
 

--- a/package/CHANGELOG
+++ b/package/CHANGELOG
@@ -13,14 +13,15 @@ The rules for this file:
   * release numbers follow "Semantic Versioning" http://semver.org
 
 ------------------------------------------------------------------------------
-??/??/?? IAlibay, melomcr, mdd31, ianmkenney
+??/??/?? IAlibay, melomcr, mdd31, ianmkenney, orbeckst, richardjgowers
 
  * 2.1.0
 
 Fixes
   * Fix ITPParser charge reading (Issue #3419).
   * TRRWriter preserves timestep data (Issue #3350).
-  
+  * documentation fixes (Issues #3224, #2667)
+
 Enhancements
 
 Changes

--- a/package/MDAnalysis/analysis/density.py
+++ b/package/MDAnalysis/analysis/density.py
@@ -102,13 +102,6 @@ atomgroup.
    :members:
    :inherited-members: run
 
-   .. attribute:: results.density
-
-      After the analysis (see the :meth:`~DensityAnalysis.run` method), the
-      resulting density is stored in the :attr:`results.density` attribute as
-      a :class:`Density` instance. Note: this replaces the now deprecated
-      :attr:`density` attribute.
-
    .. automethod:: _set_user_grid
 
 
@@ -216,6 +209,11 @@ class DensityAnalysis(AnalysisBase):
     results.density : :class:`Density`
             A :class:`Density` instance containing a physical density of units
             :math:`Angstrom^{-3}`.
+
+            After the analysis (see the :meth:`~DensityAnalysis.run` method),
+            the resulting density is stored in the :attr:`results.density`
+            attribute as a :class:`Density` instance. Note: this replaces the
+            now deprecated :attr:`density` attribute.
 
     density : :class:`Density`
             Alias to the :attr:`results.density`.
@@ -396,6 +394,7 @@ class DensityAnalysis(AnalysisBase):
        :func:`_set_user_grid` is now a method of :class:`DensityAnalysis`.
        :class:`Density` results are now stored in a
        :class:`MDAnalysis.analysis.base.Results` instance.
+
     """
 
     def __init__(self, atomgroup, delta=1.0,

--- a/package/MDAnalysis/analysis/encore/similarity.py
+++ b/package/MDAnalysis/analysis/encore/similarity.py
@@ -20,7 +20,8 @@
 # MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
 # J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
 #
-"""=================================================================================
+"""
+=================================================================================
 Ensemble Similarity Calculations --- :mod:`MDAnalysis.analysis.encore.similarity`
 =================================================================================
 
@@ -157,11 +158,13 @@ Functions for ensemble comparisons
 ==================================
 
 .. autofunction:: hes
+   :noindex:
 
 .. autofunction:: ces
+   :noindex:
 
 .. autofunction:: dres
-
+   :noindex:
 
 Function reference
 ==================

--- a/package/MDAnalysis/analysis/hole2/hole.py
+++ b/package/MDAnalysis/analysis/hole2/hole.py
@@ -560,7 +560,7 @@ class HoleAnalysis(AnalysisBase):
 
     .. versionchanged:: 2.0.0
         :attr:`sphpdbs`, :attr:`outfiles` and :attr:`profiles `
-        are now stored in a :class:`MDAnalysis.analysis.base.Results` 
+        are now stored in a :class:`MDAnalysis.analysis.base.Results`
         instance.
 
     """
@@ -1300,14 +1300,14 @@ class HoleAnalysis(AnalysisBase):
             Profiles to include by frame. If ``None``, includes
             all frames.
         bins : int or iterable of edges, optional
-            If bins is an int, it defines the number of equal-width bins in the given 
-            range. If bins is a sequence, it defines a monotonically increasing array of 
+            If bins is an int, it defines the number of equal-width bins in the given
+            range. If bins is a sequence, it defines a monotonically increasing array of
             bin edges, including the rightmost edge, allowing for non-uniform bin widths.
         range : (float, float), optional
             The lower and upper range of the bins.
             If not provided, ``range`` is simply ``(a.min(), a.max())``,
             where ``a`` is the array of reaction coordinates.
-            Values outside the range are ignored. The first element of the range must be 
+            Values outside the range are ignored. The first element of the range must be
             less than or equal to the second.
 
 
@@ -1358,14 +1358,14 @@ class HoleAnalysis(AnalysisBase):
             Profiles to include by frame. If ``None``, includes
             all frames.
         bins : int or iterable of edges, optional
-            If bins is an int, it defines the number of equal-width bins in the given 
-            range. If bins is a sequence, it defines a monotonically increasing array of 
+            If bins is an int, it defines the number of equal-width bins in the given
+            range. If bins is a sequence, it defines a monotonically increasing array of
             bin edges, including the rightmost edge, allowing for non-uniform bin widths.
         range : (float, float), optional
             The lower and upper range of the bins.
             If not provided, ``range`` is simply ``(a.min(), a.max())``,
             where ``a`` is the array of reaction coordinates.
-            Values outside the range are ignored. The first element of the range must be 
+            Values outside the range are ignored. The first element of the range must be
             less than or equal to the second.
 
 
@@ -1394,14 +1394,14 @@ class HoleAnalysis(AnalysisBase):
             Profiles to include by frame. If ``None``, includes
             all frames.
         bins : int or iterable of edges, optional
-            If bins is an int, it defines the number of equal-width bins in the given 
-            range. If bins is a sequence, it defines a monotonically increasing array of 
+            If bins is an int, it defines the number of equal-width bins in the given
+            range. If bins is a sequence, it defines a monotonically increasing array of
             bin edges, including the rightmost edge, allowing for non-uniform bin widths.
         range : (float, float), optional
             The lower and upper range of the bins.
             If not provided, ``range`` is simply ``(a.min(), a.max())``,
             where ``a`` is the array of reaction coordinates.
-            Values outside the range are ignored. The first element of the range must be 
+            Values outside the range are ignored. The first element of the range must be
             less than or equal to the second.
         color : str or array-like, optional
             Color for the plot.

--- a/package/MDAnalysis/analysis/pca.py
+++ b/package/MDAnalysis/analysis/pca.py
@@ -208,14 +208,6 @@ class PCA(AnalysisBase):
                 Will be removed in MDAnalysis 3.0.0. Please use
                 :attr:`results.cumulated_variance` instead.
 
-
-    Methods
-    -------
-    transform(atomgroup, n_components=None)
-        Take an atomgroup or universe with the same number of atoms as was
-        used for the calculation in :meth:`PCA.run`, and project it onto the
-        principal components.
-
     Notes
     -----
     Computation can be sped up by supplying precalculated mean positions.

--- a/package/MDAnalysis/coordinates/FHIAIMS.py
+++ b/package/MDAnalysis/coordinates/FHIAIMS.py
@@ -63,10 +63,6 @@ specifications in the official `FHI-AIMS format`_.
 Classes
 -------
 
-.. autoclass:: Timestep
-   :members:
-   :inherited-members:
-
 .. autoclass:: FHIAIMSReader
    :members:
    :inherited-members:

--- a/package/MDAnalysis/coordinates/GRO.py
+++ b/package/MDAnalysis/coordinates/GRO.py
@@ -52,9 +52,6 @@ For example::
 Classes
 -------
 
-.. autoclass:: Timestep
-   :members:
-
 .. autoclass:: GROReader
    :members:
 

--- a/package/MDAnalysis/coordinates/TRZ.py
+++ b/package/MDAnalysis/coordinates/TRZ.py
@@ -38,45 +38,6 @@ trajectories in *little-endian* byte order.
 Classes
 -------
 
-.. autoclass:: MDAnalysis.coordinates.TRZ.Timestep
-   :members:
-
-   .. attribute:: frame
-
-      Index of current frame number (0 based)
-
-   .. attribute:: time
-
-      Current system time in ps
-
-   .. attribute:: n_atoms
-
-      Number of atoms in the frame (will be constant throughout trajectory)
-
-   .. attribute:: pressure
-
-      System pressure in pascals
-
-   .. attribute:: pressure_tensor
-
-      Array containing pressure tensors in order: xx, xy, yy, xz, yz, zz
-
-   .. attribute:: total_energy
-
-      Hamiltonian for the system in kJ/mol
-
-   .. attribute:: potential_energy
-
-      Potential energy of the system in kJ/mol
-
-   .. attribute:: kinetic_energy
-
-      Kinetic energy of the system in kJ/mol
-
-   .. attribute:: temperature
-
-      Temperature of the system in Kelvin
-
 .. autoclass:: TRZReader
    :members:
 
@@ -101,8 +62,8 @@ class TRZReader(base.ReaderBase):
 
     Attributes
     ----------
-    ts : TRZ.Timestep
-         :class:`~MDAnalysis.coordinates.TRZ.Timestep` object containing
+    ts : Timestep
+         :class:`~MDAnalysis.coordinates.base.Timestep` object containing
          coordinates of current frame
 
     Note

--- a/package/MDAnalysis/coordinates/TRZ.py
+++ b/package/MDAnalysis/coordinates/TRZ.py
@@ -62,7 +62,7 @@ class TRZReader(base.ReaderBase):
 
     Attributes
     ----------
-    ts : Timestep
+    ts : base.Timestep
          :class:`~MDAnalysis.coordinates.base.Timestep` object containing
          coordinates of current frame
 

--- a/package/MDAnalysis/core/topology.py
+++ b/package/MDAnalysis/core/topology.py
@@ -114,7 +114,7 @@ def make_downshift_arrays(upshift, nparents):
     """
     if not len(upshift):
         return np.array([], dtype=object)
-        
+
     order = np.argsort(upshift)
 
     upshift_sorted = upshift[order]
@@ -178,8 +178,9 @@ class TransTable(object):
         number of residues in topology
     n_segments : int
         number of segments in topology
-    size
-        tuple describing the shape of the TransTable
+    size : tuple
+        tuple ``(n_atoms, n_residues, n_segments)`` describing the shape of
+        the TransTable
     """
     def __init__(self,
                  n_atoms, n_residues, n_segments,  # Size of tables
@@ -214,7 +215,10 @@ class TransTable(object):
 
     @property
     def size(self):
-        """The shape of the table, (n_atoms, n_residues, n_segments)"""
+        """The shape of the table, ``(n_atoms, n_residues, n_segments)``.
+
+        :meta private:
+        """
         return (self.n_atoms, self.n_residues, self.n_segments)
 
     def atoms2residues(self, aix):
@@ -575,4 +579,4 @@ class Topology(object):
             attr.values = np.concatenate([attr.values, np.array([newval])])
 
         return segidx
-        
+

--- a/package/MDAnalysis/core/topology.py
+++ b/package/MDAnalysis/core/topology.py
@@ -180,28 +180,6 @@ class TransTable(object):
         number of segments in topology
     size
         tuple describing the shape of the TransTable
-
-    Methods
-    -------
-    atoms2residues(aix)
-        Returns the residue index for many atom indices
-    residues2atoms_1d(rix)
-        All atoms in the residues represented by *rix*
-    residues2atoms_2d(rix)
-        List of atom indices for each residue in *rix*
-    residues2segments(rix)
-        Segment indices for each residue in *rix*
-    segments2residues_1d(six)
-        Similar to `residues2atoms_1d`
-    segments2residues_2d(six)
-        Similar to `residues2atoms_2d`
-    atoms2segments(aix)
-        Segment indices for each atom in *aix*
-    segments2atoms_1d(six)
-        Similar to `residues2atoms_1d`
-    segments2atoms_2d(six)
-        Similar to `residues2atoms_2d`
-
     """
     def __init__(self,
                  n_atoms, n_residues, n_segments,  # Size of tables

--- a/package/MDAnalysis/core/universe.py
+++ b/package/MDAnalysis/core/universe.py
@@ -409,7 +409,7 @@ class Universe(object):
         trajectory: bool, optional
           if ``True``, attaches a
           :class:`MDAnalysis.coordinates.memory.MemoryReader` allowing
-          coordinates to be set and written.  Default is ``False``
+          coordinates to be set and written.
         velocities: bool, optional
           include velocities in the
           :class:`MDAnalysis.coordinates.memory.MemoryReader`
@@ -1374,7 +1374,7 @@ class Universe(object):
             hydrogens with the `addHs` parameter
         numConfs : int (optional, default 1)
             Number of frames to generate coordinates for. Ignored if
-            `generate_coordinates` is ``False``
+            ``generate_coordinates=False``
         rdkit_kwargs : dict (optional)
             Other arguments passed to the RDKit :func:`EmbedMultipleConfs`
             function

--- a/package/MDAnalysis/core/universe.py
+++ b/package/MDAnalysis/core/universe.py
@@ -177,49 +177,33 @@ def _generate_from_topology(universe):
 class Universe(object):
     """The MDAnalysis Universe contains all the information describing the system.
 
-    The system always requires a *topology* file --- in the simplest case just
+    The system always requires a *topology file* --- in the simplest case just
     a list of atoms. This can be a CHARMM/NAMD PSF file or a simple coordinate
-    file with atom informations such as XYZ, PDB, Gromacs GRO, or CHARMM
+    file with atom informations such as XYZ, PDB, GROMACS GRO or TPR, or CHARMM
     CRD. See :ref:`Supported topology formats` for what kind of topologies can
     be read.
 
-    A trajectory provides coordinates; the coordinates have to be ordered in
-    the same way as the list of atoms in the topology. A trajectory can be a
-    single frame such as a PDB, CRD, or GRO file, or it can be a MD trajectory
-    (in CHARMM/NAMD/LAMMPS DCD, Gromacs XTC/TRR, or generic XYZ format).  See
-    :ref:`Supported coordinate formats` for what can be read as a
-    "trajectory".
+    A *trajectory file* provides coordinates; the coordinates have to be
+    ordered in the same way as the list of atoms in the topology. A trajectory
+    can be a single frame such as a PDB, CRD, or GRO file, or it can be a MD
+    trajectory (in CHARMM/NAMD/LAMMPS DCD, GROMACS XTC/TRR, AMBER nc, generic
+    XYZ format, ...).  See :ref:`Supported coordinate formats` for what can be
+    read as a "trajectory".
 
     As a special case, when the topology is a file that contains atom
     information *and* coordinates (such as XYZ, PDB, GRO or CRD, see
     :ref:`Supported coordinate formats`) then the coordinates are immediately
     loaded from the "topology" file unless a trajectory is supplied.
 
-    Examples for setting up a universe::
-
-       u = Universe(topology, trajectory)          # read system from file(s)
-       u = Universe(pdbfile)                       # read atoms and coordinates from PDB or GRO
-       u = Universe(topology, [traj1, traj2, ...]) # read from a list of trajectories
-       u = Universe(topology, traj1, traj2, ...)   # read from multiple trajectories
-
-    Load new data into a universe (replaces old trajectory and does *not* append)::
-
-       u.load_new(trajectory)                      # read from a new trajectory file
-
-    Select atoms, with syntax similar to CHARMM (see
-    :class:`~Universe.select_atoms` for details)::
-
-       u.select_atoms(...)
-
     Parameters
     ----------
-    topology: str, stream, `~MDAnalysis.core.topology.Topology`, `np.ndarray`, None
+    topology: str, stream, Topology, numpy.ndarray, None
         A CHARMM/XPLOR PSF topology file, PDB file or Gromacs GRO file; used to
         define the list of atoms. If the file includes bond information,
         partial charges, atom masses, ... then these data will be available to
         MDAnalysis. Alternatively, an existing
         :class:`MDAnalysis.core.topology.Topology` instance may be given,
-        numpy coordinates, or None for an empty universe.
+        numpy coordinates, or ``None`` for an empty universe.
     coordinates: str, stream, list of str, list of stream (optional)
         Coordinates can be provided as files of
         a single frame (eg a PDB, CRD, or GRO file); a list of single
@@ -272,15 +256,53 @@ class Universe(object):
 
     Attributes
     ----------
-    trajectory
-        currently loaded trajectory reader;
-    dimensions
-        current system dimensions (simulation unit cell, if set in the
-        trajectory)
-    atoms, residues, segments
-        principal Groups for each topology level
-    bonds, angles, dihedrals
-        principal ConnectivityGroups for each connectivity type
+    trajectory : base.ReaderBase or base.SingleFrameReaderBase
+        currently loaded trajectory reader; readers are described in
+        :ref:`Coordinates`
+    dimensions : numpy.ndarray
+        system dimensions (simulation unit cell, if set in the
+        trajectory) at the *current time step*
+        (see :attr:`MDAnalysis.coordinates.base.Timestep.dimensions`).
+        The unit cell can be set for the current time step (but the change is
+        not permanent unless written to a file).
+    atoms : AtomGroup
+        all particles (:class:`~MDAnalysis.core.groups.Atom`) in the system,
+        as read from the `topology` file
+    residues : ResidueGroup
+        all residues (:class:`~MDAnalysis.core.groups.Residue`) in the system
+    segments : SegmentGroup
+        all segments (:class:`~MDAnalysis.core.groups.Segment`) in the system
+    bonds : topologyattrs.Bonds
+        all bonds (if defined in the `topology`) as provided by
+        :attr:`Universe.atoms.bonds`
+    angles : topologyattrs.Angles
+        all angles (if defined in the `topology`), same as
+        :attr:`Universe.atoms.angles`
+    dihedrals : topologyattrs.Dihedrals
+        all dihedral angles (if defined in the `topology`), same as
+        :attr:`Universe.atoms.dihedrals`
+    impropers : topologyattrs.Impropers
+        all improper dihedral angles (if defined in the `topology`), same as
+        :attr:`Universe.atoms.impropers`
+
+    Examples
+    --------
+    Examples for setting up a :class:`Universe`::
+
+       u = Universe(topology, trajectory)          # read system from file(s)
+       u = Universe(pdbfile)                       # read atoms and coordinates from PDB or GRO
+       u = Universe(topology, [traj1, traj2, ...]) # read from a list of trajectories
+       u = Universe(topology, traj1, traj2, ...)   # read from multiple trajectories
+
+    Load new data into a universe (replaces old trajectory and does *not* append)::
+
+       u.load_new(trajectory)                      # read from a new trajectory file
+
+    Selecting atoms with :meth:`~Universe.select_atoms` ::
+
+       ag = u.select_atoms(...)
+
+    returns an :class:`~MDAnalysis.core.groups.AtomGroup`.
 
 
     .. versionchanged:: 1.0.0
@@ -289,8 +311,8 @@ class Universe(object):
 
     .. versionchanged:: 2.0.0
         Universe now can be (un)pickled.
-        ``topology`` and ``trajectory`` are reserved
-        upon unpickle.
+        ``topology`` and ``trajectory`` are reserved upon unpickle.
+
     """
     def __init__(self, topology=None, *coordinates, all_coordinates=False,
                  format=None, topology_format=None, transformations=None,
@@ -366,7 +388,7 @@ class Universe(object):
         Useful for building a Universe without requiring existing files,
         for example for system building.
 
-        If `trajectory` is set to True, a
+        If `trajectory` is set to ``True``, a
         :class:`MDAnalysis.coordinates.memory.MemoryReader` will be
         attached to the Universe.
 
@@ -385,33 +407,39 @@ class Universe(object):
         residue_segindex: array like, optional
           mapping of residues to segments
         trajectory: bool, optional
-          if True, attaches a :class:`MDAnalysis.coordinates.memory.MemoryReader`
-          allowing coordinates to be set and written.  Default is False
+          if ``True``, attaches a
+          :class:`MDAnalysis.coordinates.memory.MemoryReader` allowing
+          coordinates to be set and written.  Default is ``False``
         velocities: bool, optional
-          include velocities in the :class:`MDAnalysis.coordinates.memory.MemoryReader`
+          include velocities in the
+          :class:`MDAnalysis.coordinates.memory.MemoryReader`
         forces: bool, optional
-          include forces in the :class:`MDAnalysis.coordinates.memory.MemoryReader`
+          include forces in the
+          :class:`MDAnalysis.coordinates.memory.MemoryReader`
 
         Returns
         -------
-        MDAnalysis.Universe object
+        Universe
+          :class:`~MDAnalysis.core.universe.Universe` instance with dummy
+          values for atoms and undefined coordinates/velocities/forces
 
         Examples
         --------
         For example to create a new Universe with 6 atoms in 2 residues, with
-        positions for the atoms and a mass attribute:
+        positions for the atoms and a mass attribute::
 
-        >>> u = mda.Universe.empty(6, 2,
-                                   atom_resindex=np.array([0, 0, 0, 1, 1, 1]),
-                                   trajectory=True,
+          u = mda.Universe.empty(6, 2,
+                                 atom_resindex=np.array([0, 0, 0, 1, 1, 1]),
+                                 trajectory=True,
                 )
-        >>> u.add_TopologyAttr('masses')
+          u.add_TopologyAttr('masses')
 
         .. versionadded:: 0.17.0
         .. versionchanged:: 0.19.0
            The attached Reader when trajectory=True is now a MemoryReader
         .. versionchanged:: 1.0.0
            Universes can now be created with 0 atoms
+
         """
         if not n_atoms:
             n_residues = 0
@@ -496,8 +524,9 @@ class Universe(object):
 
         Raises
         ------
-        TypeError if trajectory format can not be
-                  determined or no appropriate trajectory reader found
+        TypeError
+             if trajectory format can not be determined or no appropriate
+             trajectory reader found
 
 
         .. versionchanged:: 0.8
@@ -512,6 +541,7 @@ class Universe(object):
         .. versionchanged:: 0.17.0
            Now returns a :class:`Universe` instead of the tuple of file/array
            and detected file type.
+
         """
         # filename==None happens when only a topology is provided
         if filename is None:
@@ -629,22 +659,34 @@ class Universe(object):
 
     @property
     def bonds(self):
-        """Bonds between atoms"""
+        """Bonds between atoms.
+
+        :meta private:
+        """
         return self.atoms.bonds
 
     @property
     def angles(self):
-        """Angles between atoms"""
+        """Angles between atoms.
+
+        :meta private:
+        """
         return self.atoms.angles
 
     @property
     def dihedrals(self):
-        """Dihedral angles between atoms"""
+        """Dihedral angles between atoms.
+
+        :meta private:
+        """
         return self.atoms.dihedrals
 
     @property
     def impropers(self):
-        """Improper dihedral angles between atoms"""
+        """Improper dihedral angles between atoms.
+
+        :meta private:
+        """
         return self.atoms.impropers
 
     def __repr__(self):
@@ -674,7 +716,10 @@ class Universe(object):
     # Properties
     @property
     def dimensions(self):
-        """Current dimensions of the unitcell"""
+        """Current dimensions of the unitcell.
+
+        :meta private:
+        """
         return self.coord.dimensions
 
     @dimensions.setter
@@ -713,7 +758,10 @@ class Universe(object):
 
     @property
     def trajectory(self):
-        """Reference to trajectory reader object containing trajectory data."""
+        """Reference to trajectory reader object containing trajectory data.
+
+        :meta private:
+        """
         if self._trajectory is not None:
             return self._trajectory
         else:
@@ -753,7 +801,7 @@ class Universe(object):
            attribute to add (eg 'charges'), can also supply initial values
            using values keyword.
 
-        .. versionchanged:: 1.1.0 
+        .. versionchanged:: 1.1.0
             Now warns when adding bfactors to a Universe with
             existing tempfactors, or adding tempfactors to a
             Universe with existing bfactors.
@@ -1321,19 +1369,25 @@ class Universe(object):
         addHs : bool (optional, default True)
             Add all necessary hydrogens to the molecule
         generate_coordinates : bool (optional, default True)
-            Generate 3D coordinates using RDKit's `AllChem.EmbedMultipleConfs`
-            function. Requires adding hydrogens with the `addHs` parameter
+            Generate 3D coordinates using RDKit's
+            :func:`AllChem.EmbedMultipleConfs` function. Requires adding
+            hydrogens with the `addHs` parameter
         numConfs : int (optional, default 1)
             Number of frames to generate coordinates for. Ignored if
-            `generate_coordinates=False`
+            `generate_coordinates` is ``False``
         rdkit_kwargs : dict (optional)
-            Other arguments passed to the RDKit `EmbedMultipleConfs` function
+            Other arguments passed to the RDKit :func:`EmbedMultipleConfs`
+            function
         kwargs : dict
             Parameters passed on Universe creation
 
         Returns
         -------
-        :class:`~MDAnalysis.core.Universe`
+        universe : Universe
+            contains atom names and topology information (bonds) derived from
+            the input SMILES string; coordinates are included if
+            `generate_coordinates` was set to ``True``
+
 
         Examples
         --------
@@ -1354,6 +1408,7 @@ class Universe(object):
 
 
         .. versionadded:: 2.0.0
+
         """
         try:
             from rdkit import Chem

--- a/package/MDAnalysis/lib/mdamath.py
+++ b/package/MDAnalysis/lib/mdamath.py
@@ -25,10 +25,17 @@
 Mathematical helper functions --- :mod:`MDAnalysis.lib.mdamath`
 ===============================================================
 
-Helper functions for common mathematical operations
+
+Helper functions for common mathematical operations. Some of these functions
+are written in C/cython for higher performance.
+
+Linear algebra
+--------------
 
 .. autofunction:: normal
 .. autofunction:: norm
+.. autofunction:: pdot
+.. autofunction:: pnorm
 .. autofunction:: angle
 .. autofunction:: dihedral
 .. autofunction:: stp
@@ -36,12 +43,19 @@ Helper functions for common mathematical operations
 .. autofunction:: triclinic_box
 .. autofunction:: triclinic_vectors
 .. autofunction:: box_volume
+
+
+Connectivity
+------------
+
 .. autofunction:: make_whole
 .. autofunction:: find_fragments
+
 
 .. versionadded:: 0.11.0
 .. versionchanged: 1.0.0
    Unused function :func:`_angle()` has now been removed.
+
 """
 import numpy as np
 

--- a/package/doc/sphinx/source/documentation_pages/analysis/hole2.rst
+++ b/package/doc/sphinx/source/documentation_pages/analysis/hole2.rst
@@ -21,7 +21,7 @@ Module
 ------
 
 .. automodule:: MDAnalysis.analysis.hole2.hole
-    :members:
+		
 
 
 Utility functions and templates

--- a/package/doc/sphinx/source/documentation_pages/coordinates/TRZ.rst
+++ b/package/doc/sphinx/source/documentation_pages/coordinates/TRZ.rst
@@ -1,2 +1,2 @@
 .. automodule:: MDAnalysis.coordinates.TRZ
-   :members:
+

--- a/package/doc/sphinx/source/documentation_pages/lib/mdamath.rst
+++ b/package/doc/sphinx/source/documentation_pages/lib/mdamath.rst
@@ -1,2 +1,1 @@
 .. automodule:: MDAnalysis.lib.mdamath
-   :members:

--- a/package/doc/sphinx/source/documentation_pages/selections/charmm.rst
+++ b/package/doc/sphinx/source/documentation_pages/selections/charmm.rst
@@ -1,4 +1,2 @@
 .. automodule:: MDAnalysis.selections.charmm
-   :members:
-   :inherited-members:
 

--- a/package/doc/sphinx/source/documentation_pages/selections/gromacs.rst
+++ b/package/doc/sphinx/source/documentation_pages/selections/gromacs.rst
@@ -1,3 +1,1 @@
 .. automodule:: MDAnalysis.selections.gromacs
-   :members:
-   :inherited-members:

--- a/package/doc/sphinx/source/documentation_pages/selections/jmol.rst
+++ b/package/doc/sphinx/source/documentation_pages/selections/jmol.rst
@@ -1,3 +1,1 @@
 .. automodule:: MDAnalysis.selections.jmol
-   :members:
-   :inherited-members:

--- a/package/doc/sphinx/source/documentation_pages/selections/pymol.rst
+++ b/package/doc/sphinx/source/documentation_pages/selections/pymol.rst
@@ -1,4 +1,2 @@
 .. automodule:: MDAnalysis.selections.pymol
-   :members:
-   :inherited-members:
 

--- a/package/doc/sphinx/source/documentation_pages/selections/vmd.rst
+++ b/package/doc/sphinx/source/documentation_pages/selections/vmd.rst
@@ -1,4 +1,2 @@
 .. automodule:: MDAnalysis.selections.vmd
-   :members:
-   :inherited-members:
 

